examples/vips_class.php

```diff
@@ -1,18 +1,63 @@
 #!/usr/bin/env php
 <?php
 
-include '../src/Image.php';
+require __DIR__ . '/../vendor/autoload.php';
 
 use Jcupitt\Vips;
 
-Vips\Image::setLogging(true);
-
-$image = Vips\Image::newFromFile($argv[1]); 
-
-echo "width = ", $image->width, "\n";
+const LOG_FORMAT = "[%datetime%] %level_name%: %message% %context%\n";
+const DATE_FORMAT = "Y-m-d\TH:i:sP";
+
+Vips\Image::setLogger(new class implements Psr\Log\LoggerInterface {
+    // Use the LoggerTait so that we only have to implement the generic log method.
+    use Psr\Log\LoggerTrait;
+
+    /**
+     * Logs with an arbitrary level.
+     *
+     * @param mixed  $level
+     * @param string $message
+     * @param array  $context
+     *
+     * @return void
+     */
+    public function log($level, $message, array $context = [])
+    {
+        // `Vips\Image` to string convert
+        array_walk_recursive($context, function (&$value) {
+            if ($value instanceof Vips\Image) {
+                // TODO: Should we override __toString() in `Vips\Image` instead
```

Comment:
I forget to include this `TODO` in the pull request description. I've implemented some basic `__toString()` in here, maybe we should put this in the `Image` class to let users use `$imageText = (string)$image;`.